presto-hive-metastore/src/main/java/com/facebook/presto/hive/PartitionExtraInfoFetcher.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import org.apache.hadoop.hive.metastore.api.Partition;
+
+import java.util.Optional;
+
+public interface PartitionExtraInfoFetcher
+{
+    default Optional<Integer> getPartitionVersion(Partition partition)
```

Comment:
We don't have to make this a default method since the Hive implementation would be the default :)